core/Period/Factory.php

```diff
@@ -55,6 +56,17 @@ public static function build($period, $date, $timezone = 'UTC')
                 return new Year($date);
                 break;
         }
+
+        $customPeriodFactories = StaticContainer::get('period.custom.factories');
```

Comment:
Maybe we could name it like just `periods` or so and then move all hard coded ones out of here into the config?